Feature/2551 add snowflake connector

yuuteng opened the pull request:
Snowflake connector init pull request

yuuteng committed:
Add Snowflake JDBC Connector

yuuteng committed:
Update trino snapshot version to 372

yuuteng committed:
Update format of the doc of snowflake

yuuteng committed:
Update trino jdbc library import

yuuteng committed:
Fix date formatter from yyyy to uuuu

yuuteng committed:
Fix date test case

yuuteng committed:
Remove defunct property allow-drop-table

yuuteng committed:
Update trino version to 374

yuuteng committed:
Update snowflake config to adapt 374

yuuteng committed:
Update the range of the test of Date type

yuuteng committed:
Update to version 375

yuuteng committed:
Fix snowflake after updating to 375

findepi committed:
Update Iceberg table statistics on writes

yuuteng committed:
Update to 381

yuuteng committed:
Fix mvn pom import

yuuteng committed:
Format snowflake.rst

yuuteng committed:
Reorderd Data tests in type mapping

yuuteng committed:
Update function code

yuuteng committed:
Add product test

yuuteng committed:
Rename product test tablename

yuuteng committed:
Add Env, Suite and properties of Snowflake for production test

yuuteng committed:
Add trinoCreateAndInsert()

yuuteng committed:
Refactor snowflake from single node to multi node

yuuteng committed:
Pass product tests

yuuteng committed:
Removed snowflake.properties in trino server dev

yuuteng committed:
Resolved issues 19 05 2022 and fixed tests

yuuteng committed:
Remove Types.VARBINARY

yuuteng committed:
Add private static SliceWriteFunction charWriteFunction

yuuteng committed:
Update test case

yuuteng committed:
Update plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

Co-authored-by: Yuya Ebihara <ebyhry@gmail.com>

yuuteng committed:
Update docs/src/main/sphinx/connector/snowflake.rst

Co-authored-by: Yuya Ebihara <ebyhry@gmail.com>

yuuteng committed:
Update plugin/trino-snowflake/pom.xml

Co-authored-by: Yuya Ebihara <ebyhry@gmail.com>

yuuteng committed:
Update plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

Co-authored-by: Yuya Ebihara <ebyhry@gmail.com>

yuuteng committed:
Resolved review open issues

yuuteng committed:
Disabled JDBC_TREAT_DECIMAL_AS_INT and fixed test case

yuuteng committed:
Updated properties file

yuuteng committed:
Updated properties files

yuuteng committed:
Renamed properties in Testing

yuuteng committed:
Revert "Renamed properties in Testing"

This reverts commit 82f9eb3f3811e8d90a482f5359e98e7c729afa17.

yuuteng committed:
Renamed properties and fixed tests

yuuteng committed:
Update the way to pass ENV values for production test

yuuteng committed:
Update trino version to 388

yuuteng committed:
Update Trino version 391

ebyhr committed:
Add support for migrate procedure in Iceberg

yuuteng committed:
Update trino version to 394

yuuteng committed:
Update to 395

pettyjamesm committed:
Update airlift units to 1.8

Updates airlift units to 1.8 for all connector and server modules,
leaving trino-cli, trino-client, and trino-jdbc at version 1.7
since those modules still support JDK 8.

elonazoulay committed:
Minor cleanup for PinotTypeConverter

Fix comments and imports.

elonazoulay committed:
Remove redundant column in BasePinotIntegrationConnectorSmokeTest

elonazoulay committed:
Remove redundant requireNonNull check for PinotConfig

For reference see the following commit:
"Remove redundant requireNonNull when using config objects"

ebyhr committed:
Simplify predicates involving year function

findepi committed:
Rename test class to contain more scenarios

Iceberg connector will collect stats during writes, and it would be good
to test this together with ANALYZE. Rename class to contain all related
functionalities.

findepi committed:
Add last fresh time to MaterializedViewFreshness

Let connector inform the engine how fresh the materialized view is. This
is required to implement materialized view handling as per the new GRACE
PERIOD syntax definition.

Note: the property is called "last fresh time" and not "last refresh
time". This is in support for connectors which know when view became no
longer fresh. For example, a materialized view may have been refreshed a
day ago, but only an hour ago one of its base tables changed. Then the
materialized view is stale and "last fresh time" is one hour ago, not
one day ago.

findepi committed:
Respect GRACE PERIOD when querying stale mateiralized views

findepi committed:
Fix materialized view on Iceberg table with no snapshot

Before the change, a materialized view on an Iceberg table with no
snapshot would be considered always stale.

The change is hard to test, because both Trino and Spark currently
create an empty snapshot when creating a table, so such situation is
hard to get into. Older Trino version used to create tables with no
initial snapshot though.

findinpath committed:
Add CI infrastructure for `trino-redshift`

Use an ephemeral/throw-away Redshift cluster for running integration tests
on `trino-redshift` connector.
Once the tests are run, the testing Redshift cluster is being reclaimed.

The Redshift cluster is publicly accessible in order to be accessible
from the general purpose Github runners.

findepi committed:
Add grace period in ConnectorMaterializedViewDefinition

Expose the GRACE PERIOD in `ConnectorMaterializedViewDefinition` so that
connectors can store the information.

findepi committed:
Refactor testFederatedMaterializedView

- exit early when MV not supported
- add assert on freshness on initial view
- capture expected results variables -- this is preparation step for
  adding more assertions

findepi committed:
Allow mocking session time in tests

Tests operate on `Session` which has the `start` field, but this does
not get passed to the server via the `ClientSession`. Inject testing
session time provider that will allow the tests set a different session
time for components that need that for tests.

dain committed:
Add native OpenX JSON format

findinpath committed:
Bump redshift-jdbc42 to 2.1.0.12

findepi committed:
Support REFRESH MATERIALIZED VIEW when storage differs in schema

The materialized view's storage table may have a schema different than
the materialized view itself. This is already supported when reading
(the coercions are getting applied as necessary), but was not supported
for REFRESH. In the REFRESH normal INSERT constraints where getting
applied, thus not allowing the materialized view to, for example, store
temporal values in a varchar column.

The change removes the constraint, and therefore a storage table may use
any type that is coercible to from the view definition type. Of course,
the implementations should only use types where coercion to the storage
and backwards does round trip.

findepi committed:
Remove redundant format call

Per code style.

findepi committed:
Sync test relation name with test name

findepi committed:
Support arbitrary types in Iceberg materialized view definition

A materialized view should be able to capture any query results and
these are not constrained to be limited to types directly supported by
Iceberg.

Coerce unsupported types to supported ones.

When reading, when the storage table has types different than MV output
types, coercions are already applied by the `StatementAnalyzer`.

The MV/storage schema mismatch is not supported in REFRESH MATERIALIZED
VIEW yet.

pettyjamesm committed:
Add states for tasks terminating but not yet finished

This is a very large commit, but unfortuantely it's an all or nothing
change that fundamentally changes how tasks report their state
throughout the engine on both workers and on the coordinator. Before
this change, tasks would immediately change to some final state when
cancelled, failed, or told to abort- even while drivers were still
running. Additionally, the coordinator would forcibly set the task state
locally during an abort or some kinds of failure without the remote
worker acknowledging that command.

The result was that the coordinators view of the worker state was
inconsistent for the purposes of scheduling (since drivers for tasks
that were told to terminate may still be running for some amount of
time), and the final task stats may have been incomplete.

This change introduces CANCELING, ABORTING, and FAILING states that are
considered "terminating" but not yet "done", and awaiting the final
task driver to stop executing before transition to their corresponding
"done" state.

pajaks committed:
Add statistics collection for added column

pettyjamesm committed:
Include QueryId in DistributedQueryRunner query assertions

ebyhr committed:
Support inserting into Delta Lake table having invariants

ebyhr committed:
Rename variable in DeltaLakeSchemaSupport

albericgenius committed:
Add a metadata table showing commits in Hudi connector

pettyjamesm committed:
Avoid redundant TaskHandle destruction calls

Avoids unnecessary redundant TaskExecutor scheduling work when
TaskExecutor#removeTask is called more than once as a result of
terminating and done task state transition races.

pettyjamesm committed:
Use method reference lambda in SqlTask

lukasz-stec committed:
Get stats only for necessary columns

HiveTableHandle contains the list of projected columns. Since only
projected columns will be available to the engine,
HiveMetadata#getTableStatistics can return statistics limited to those
columns. This also requires CachingHiveMetastore to support caching
statistics for a subset of table columns.

Praveen2112 committed:
Extract BaseTestHiveCoercion from TestHiveCoercion

And rename TestHiveCoercion to TestHiveCoercionOnPartitionedTable

Praveen2112 committed:
Add coercion test for unpartitioned orc tables in hive

ebyhr committed:
Test query table function with column alias in BigQuery and Cassandra

ebyhr committed:
Fix failure when query table function contains column alias

MySQL, Druid, MariaDB and SingleStore connectors caused query failure.

Commit:
Support Iceberg refs system table

blvckcoffee committed:
Add version requirement for Hudi connector

pajaks committed:
Replace usage of hadoop.fs.Path with String in Delta Lake

pettyjamesm committed:
Minimize synchronization in DriverFactory

sahoss committed:
Check primary_key exist during Ignite create table

Adds a check for create table in IgniteClient. The check ensures
that the column names mentioned in the table property 'primary_key'
exists in the column defined in the create table statement.

sahoss committed:
Remove unreachable code from IgniteClient

Exception was unreachable as the same check is also present
from create table task

arhimondr committed:
Store operator summaries in MysqlEventListener

radek-kondziolka committed:
Limit number of tasks for writing stages

Limit number of tasks that will be scheduled to execute writing stages
basing on needs of specific connector or global limit -
query.max-writer-nodes-count.

Added an option query.max-writer-nodes-count to QueryManagerConfig and
a session option that limits number of tasks that take part in writing
nodes.

pettyjamesm committed:
Eagerly discard encryption keys in FileSingleStreamSpiller

arhimondr committed:
Create trino_queries table in UTF-8

blvckcoffee committed:
Remove lang codes from URLs

radek-kondziolka committed:
Add partitionCount parameter to the key in partitioningCacheMap

Map's key in PipelinedQueryScheduler.partitioningCacheMap is built only from
partitioningHandle parameter. If any two stages has same partitioning
but different number of partitions the key does not reflect that.
This is why key was extended with partitionCount.

blvckcoffee committed:
Reorder Azure storage systems

Commit:
Add FixedSplitSource constructors

Add a single element constructor and `emptyFixedSource` static creator.
No functional changes, just a slightly more elegant code.

kamil-rafalko committed:
Remove deprecated elasticsearch passthrough query

hashhar committed:
Simplify BaseJdbcClient constructor

Having multiple variants with different argument orders makes it difficult to use.

hashhar committed:
Make FTE test concurrency customizable

Some connector's underlying systems don't handle high concurrency very
well and it's useful to limit the concurrency of the test in order to
avoid flakiness or failures.

radek-kondziolka committed:
Add description for query.max-writer-tasks-count to docs

kasiafi committed:
Improve formatting in docs

homar committed:
Add ConnectorSession to TableFunctionProcessorProvider#getSplitProcessor

kasiafi committed:
Update table function returned types in docs

radek-kondziolka committed:
Enable colocated join by default

Praveen2112 committed:
Use record for ColumnContext

kasiafi committed:
Document argument types of table functions in developer guide

SchutteJan committed:
Deserialize partition value by physical name in Delta Lake

mosabua committed:
Add example for TRINO_PASSWORD usage

chenjian2664 committed:
Support predicate pushdown with like in Ignite

electrum committed:
Update to Jersey 2.39

kasiafi committed:
Document execution of table functions

kasiafi committed:
Use TableFunctionHandle example in docs

pajaks committed:
Add Hive tests for trailing space in location

pajaks committed:
Extract method for obtaining table location

findinpath committed:
Test reading `NULL` only `varchar` from Redshift view

Corresponding pr to Redshift JDBC driver:
https://github.com/aws/amazon-redshift-jdbc-driver/pull/75

Co-authored-by: Sasha Sheikin <myminitrue@gmail.com>

Praveen2112 committed:
Make version optional in ColumnContext

It allows us to re-use `ColumnContext` in scenarios which are independent
of hive version.

Praveen2112 committed:
Minor fixes in comment

- varchar is also a primitive
- tailing/trailing

pettyjamesm committed:
Cancel background Glue partition futures on failure

electrum committed:
Update to Airbase 134

chenjian2664 committed:
Support Ignite (inner, left, right) join pushdown

kasiafi committed:
Document argument types of table functions in user guide

sopel39 committed:
Use deterministic map in equality inference

regadas committed:
Add spilled data size to QueryStatistics

pettyjamesm committed:
Avoid extra allocation in OrderByOperator#getOutput()

electrum committed:
Use existing variable for file length

electrum committed:
Remove usages of Hadoop Path for Iceberg

electrum committed:
Deprecate MISSING_DATA error codes and cleanup error handling

ebyhr committed:
Use air.test.jvm.additional-arguments in Ignite

The existing maven-surefire-plugin plugin overrides
configurations in root pom.xml.

martint committed:
[maven-release-plugin] prepare for next development iteration

gaurav8297 committed:
Refactor PredicatePushdown test with simpler query

hashhar committed:
Use air.test.jvm.additional-arguments in BigQuery

The existing maven-surefire-plugin plugin overrides
configurations in root pom.xml.

Commit:
Move static methods to the end of the class

Commit:
Extract methods for better readability

JFR profiling will also be easier when code is split into more methods

kokosing committed:
Improve test method names

sopel39 committed:
Use lambda

kokosing committed:
Simplify assertions

kokosing committed:
Convert classes to records

Commit:
Update action-surefire-report

v1.6.0 uses a more efficient XML parsing library
that should handle all test reports generated in Trino.

pettyjamesm committed:
Make AsyncQueue executor visible to subclasses

findepi committed:
Remove outdated SPI revapi exclusions

sheajamba committed:
Update Hive catalog example name documentation

electrum committed:
Fix generation of JSON path keyvalue() IDs

Previously, separate IDs were generated for each member.

ebyhr committed:
Exclude Reshift tests by default

findepi committed:
Don't enforce top-level repo name in BaseCostBasedPlanTest

Engineers will typically check out the repo into a local folder named
the same way: `trino`. This, however, doesn't work when an engineer
needs to do a second checkout, or wants to make a backup of the repo for
whatever purposes. The build and tests should not assume any particular
name of the folder the repo is checked out into.

ebyhr committed:
Throw TableNotFoundException when SQLServer error is 208

ebyhr committed:
Uppercase 'from' in SQLServer getCaseSensitivityForColumns

ebyhr committed:
Separate CI job for Redshift connector

findepi committed:
Remove bogus overload in tests

The `rubixEnabledConfig` was ignored.

sopel39 committed:
Fix variable name

findepi committed:
Add Preconditions in io.trino.spi.connector package

Laonel committed:
Improve error message for catalog errors when listing metadata

Depending on the client that initiated a metadata query, error messages
returned from `ConnectorMetadata` can be confusing for users. For
example, a query to `system.jdbc` tables with no catalog predicate, on
exceptions in the catalog, will propagate those exceptions, with no way
for the user to know which catalog has thrown that exception.

When there are a lot of catalogs, this makes investigating the issue
very hard for Trino admins. We can make it easier by always including
the problematic catalog's name in the error message.

Commit:
Update action-surefire-report to fix processing large report files

mwd410 committed:
Fix BaseFailureRecoveryTest temp table checking flakiness

electrum committed:
Avoid CI log spam when downloading dependencies

kokosing committed:
Allow ViewExpression use session user explicitly

Before the change if access control is not returning any dedicated user
to evaluate
the view expression they just pass the session user. However for the
engine it is not clear that is a session user and so engine needs to
retrieve groups for that user again and possibly some session roles are
lost.

After this change access control may decide to return empty identity.
That would be in line of view SECURITY INVOKER. Then engine can simply
reuse the session identity.

nineinchnick committed:
Avoid CI log spam when downloading dependencies

chenjian2664 committed:
Add test for SharedResource

phd3 committed:
Avoid deserialization when no columns are needed

This improves performance for count(const) or count(*) queris
on formats that use GenericHiveRecordCursor.

elonazoulay committed:
Fix Pinot Timeboundary handling

Commit:
Correct misuse of Description annotation in Config

- Within Config, @ConfigDescription should be used, not @Description of spi

Praveen2112 committed:
Support coercing char types in hive connector

Laonel committed:
Add listTablePrivileges to MockConnector

lukasz-stec committed:
Extract common empty projectedColumns constructor

lukasz-stec committed:
Populate all columns in HiveTableHandle#projectedColumns by default

Before applyProjection is called or when projection pushdown is disabled,
we should assume that all table columns are to be projected for populating
HiveTableHandle#projectedColumns

findepi committed:
Get bytes from ByteBuffer safely

`ByteBuffer.array()` is a convenient and efficient way to get bytes from
a `ByteBuffer`. It has, however, numerous preconditions that should be
checked before using the returned array. The commit replaces all
`ByteBuffer.array()` usages where these preconditions are assumed to be
true.

wendigo committed:
Expose JMX RMI in product tests

This allows to connect to the running Trino instance using JDK Mission Control
and capture Java Flight Recorder recordings.

kasiafi committed:
Do not inline non-deterministic expressions as projection

krvikash committed:
Fix FilesTable for non-null key_metadata, split_offsets, equality_ids

Co-authored-by: 0xffmeta <jqy3222089@gmail.com>

findepi committed:
Add clarification javadoc to non-spilling Join operators

findepi committed:
Improve assertions in testCleaningUpWithTableWithSpecifiedLocation

- prefer `assertThat(List).hasSize` to `assertThat(list.size())` as the
  former includes list's elements upon failure,
- drop table after test.

findepi committed:
Fix metadata file listing in testCleaningUpWithTableWithSpecifiedLocation

Before the change, the test used `getAllMetadataFilesFromTableDirectory`
utility, but it was listing all (metadata and data) files. Similar
method, `getAllMetadataFilesFromTableDirectoryForTable`, listing only
metadata files existed, but was not used in the test.

The change merges the utility methods: the correct logic comes from
`getAllMetadataFilesFromTableDirectoryForTable` (so no behavior change
for tests other than `testCleaningUpWithTableWithSpecifiedLocation`),
and the name comes from `getAllMetadataFilesFromTableDirectory`.

kokosing committed:
Log impacted features

This is helpful in CI to understand the context on how tests are
executed.

Impacted features are passed as a log file and then they are passed from
suite to test run, so logging them here make sure if everything is
passed correctly.

wendigo committed:
Increase stack depth reporting for JFR

findepi committed:
Remove unnecessary parenthesis

They do not improve readability.

findepi committed:
Suppress error-prone for pointer-like javadocs

dain committed:
Do not verify length when detecting if a compressed file is split

For legacy reasons, line oriented files may be replaced during query
execution.  The new file may have a different length, so avoid checking
that the length is consistent.

Commit:
Improve documentation around impersonation

electrum committed:
Remove shaded JAXB and cleanup IDEA hack for Phoenix

dain committed:
Handle Hive isSplittable directly for built in formats

dain committed:
Fix header and footer validation for line page source

Do not check length when detecting a split file, because Hive allows
files to be replaced during a query.  Instead only the start position is
checked.
Allow a single header line for split files, because this always works
due to the way file split handling works.

Commit:
Add regex to Hive table properties format row

ebyhr committed:
Add sinh function to return hyperbolic sine

ebyhr committed:
Remove unused methods from DeltaLakeTableHandle

raunaqmorarka committed:
Simplify column adaptations for parquet in iceberg

Avoids having two different column adaptation layers
for parquet in iceberg
Makes ParquetPageSource more consistent with OrcPageSource

raunaqmorarka committed:
Fix formatting in TestTimestampMicros

pettyjamesm committed:
Reset interrupted flag in KuduSplitManager

pettyjamesm committed:
Reset interrupted flag in PinotLegacyDataFetcher

pettyjamesm committed:
Reset interrupted flag in legacy raptor shard manager and recorder

linzebing committed:
Close its source exchanges once a stage finishes execution

mosabua committed:
Improve docs for skewness function

electrum committed:
Remove usage of Hive constant from Iceberg MigrateProcedure

electrum committed:
Fix build failure due to units upgrade

aislinnjeske committed:
Upgrade jsonwebtoken to 0.11.5

albericgenius committed:
Move buildTableMetaClient to HudiUtil class

sopel39 committed:
Fix spelling mistake

findepi committed:
Pull schema access out of loop

radek-kondziolka committed:
Fix validation of session variable MAX_HASH_PARTITION_COUNT

ebyhr committed:
Restore skip archive handler in Iceberg Glue catalog

ce42133e1f broke a logic to skip archive in Iceberg Glue catalog.

findepi committed:
Adjust class name to follow convention

All Iceberg connector test classes match `TestIceberg*ConnectorTest`
pattern.

findepi committed:
Pick nicer test values

findepi committed:
Avoid initializing expected to null

findepi committed:
Remove redundant catch in a test

pajaks committed:
Refactor setting session EXTENDED_STATISTICS_COLLECT_ON_WRITE property

ebyhr committed:
Add REFS to TestIcebergGlueCatalogAccessOperations.testSelectSystemTable

chenjian2664 committed:
Fix example query in MERGE docs

findepi committed:
Avoid loading lineitem unnecessarily

losipiuk committed:
Fix stage scheduling logic in FTE mode

findepi committed:
Remove Iceberg connector tests for REST and JDBC catalogs

Like for Glue, we have connector *smoke* tests for these.

This also removes `additional-catalog-tests` profile, as the remaining
tests are short enough not to warrant a separate profile on the CI
(below 2:30 minutes).

pajaks committed:
Add clean up for DeltaLake ANALYZE tests

findepi committed:
Include checked arguments in checks messages

findepi committed:
Remove unused return values

findepi committed:
Fix hashing in BlackHoleNodePartitioningProvider

It ignored `position` and didn't check for null, so could fail.

Even if this doesn't matter for blackhole connector, the code could be
used as inspiration for other bucketing function implementations.

electrum committed:
Remove unused method from QueryManagerStats

hashhar committed:
Open SingleStoreClient for extension

losipiuk committed:
Set maxOutputLength correctly

Previously we were not accounting for bytesPreserved.
With recent check added to aircompressor code
(https://github.com/airlift/aircompressor/commit/127b7f3c6330e6cc595f10648b68a197974b50bf)
it started manifesting as java.lang.IllegalArgumentException.

This is strictly validation fix. The passed array was not filled beyond
allocated size.

pajaks committed:
Move resources path resolution to test case

homar committed:
Prevent Hive from returning table handles for Hudi table

krvikash committed:
Use Files.writeString API to write string in file

krvikash committed:
Remove redundant call to toIntExact

krvikash committed:
Remove redundant throws clause

krvikash committed:
Replace the deprecated method getRootCause with rootCause method

krvikash committed:
Replace the deprecated method getCause with cause method

findepi committed:
Print diagnostic messages to stderr when testing docker image

Praveen2112 committed:
Replace usage of Arrays#asList with ImmutableList#of

For list with null values we still use Arrays#asList

sopel39 committed:
Allow to disable transactional directory listing cache

ssheikin committed:
Fix BigQuery tests for column names with special characters

Some column names started to fail because of BigQuery internal change
~ between
18:00 16 of March 2023 GMT and
00:00 17 of March 2023 GMT,
first noticed when 90d6f0737a1f8dd4fbb8b85c2d95b20093d1b4ba merged to
master.

findepi committed:
Remove duplicate code branch

The null case is handled 3 lines below.

sopel39 committed:
Improve LongSumAggregation and DoubleSumAggregation serde

Previously row(bigint, boolean, bigint, boolean)] was used
for long sum serde (and similar for doubles).

Now it's row(bigint, bigint)

The extra bigint (row count) is still only required for WINDOW function
but this PR slighly improves serde.

We can remove extra booleans because Long and Decimal aggregations
use row count to determine if null should be outputted.

sopel39 committed:
Make hive.metastore-stats-cache-ttl >= to hive.metastore-cache-ttl

Some deployments might had hive.metastore-cache-ttl already set.
https://github.com/trinodb/trino/pull/15811 introduced new config
hive.metastore-stats-cache-ttl which could be lower (by default)
for such deployments. However, hive.metastore-cache-ttl
should take precedense over hive.metastore-stats-cache-ttl as
it's more generic (affects whole metastore cache).

ebyhr committed:
Fix serialization issue in MongoDB query function

ebyhr committed:
Add generated expression to extra info in Delta Lake

ebyhr committed:
Add test for identity column in Delta Lake connector

findepi committed:
Avoid boxing in DeltaLakeBucketFunction

Align hashing implementation with the one in
`BlackHoleNodePartitioningProvider`.

findepi committed:
Fix BaseFailureRecoveryTest temp table checking message pattern

The expected exception message pattern needs to match the one used in
`StatementAnalyzer.Visitor#visitTable`, e.g. the table name should be
expected in apostrophes.

This should fix errors on CI like the one below

```
Error:  io.trino.plugin.bigquery.TestBigQueryTaskFailureRecoveryTest.testParallel[testExplainAnalyze](9)  Time elapsed: 107.3 s  <<< FAILURE!
java.lang.AssertionError:
[There should be no remaining tmp_trino tables that are queryable. They are:
	For queryId [20230320_124434_00162_6w4f2] (prefix [tmp_trino_3a3f1a6f_]) remaining tables: [tmp_trino_3a3f1a6f_e3d45e3f]
		With errors: [
			Expecting throwable message:
			  "line 1:15: Table 'bigquery.tpch.tmp_trino_3a3f1a6f_e3d45e3f' does not exist"
			to contain:
			  "tmp_trino_3a3f1a6f_e3d45e3f does not exist"
			but did not.

			Throwable that failed the check:

			io.trino.testing.QueryFailedException: line 1:15: Table 'bigquery.tpch.tmp_trino_3a3f1a6f_e3d45e3f' does not exist
				at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
				at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:85)
				at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:477)
				at io.trino.testing.BaseFailureRecoveryTest.lambda$checkTemporaryTables$15(BaseFailureRecoveryTest.java:439)
				at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:63)
				at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:892)
				at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1366)
				at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1210)
				at io.trino.testing.BaseFailureRecoveryTest.checkTemporaryTables(BaseFailureRecoveryTest.java:439)
				at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.cleansUpTemporaryTables(BaseFailureRecoveryTest.java:619)
				at io.trino.testing.BaseFailureRecoveryTest.testNonSelect(BaseFailureRecoveryTest.java:365)
				at io.trino.testing.BaseFailureRecoveryTest.testTableModification(BaseFailureRecoveryTest.java:342)
				at io.trino.testing.BaseFailureRecoveryTest.testTableModification(BaseFailureRecoveryTest.java:337)
				at io.trino.testing.BaseFailureRecoveryTest.testExplainAnalyze(BaseFailureRecoveryTest.java:315)
				at io.trino.testing.BaseFailureRecoveryTest$ParallelTestRunnable.run(BaseFailureRecoveryTest.java:899)
				at io.trino.testing.BaseFailureRecoveryTest.testParallel(BaseFailureRecoveryTest.java:222)
				at jdk.internal.reflect.GeneratedMethodAccessor1309.invoke(Unknown Source)
				at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
				at java.base/java.lang.reflect.Method.invoke(Method.java:568)
				at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
				at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
				at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
				at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:75)
				at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:14)
				at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
				at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
				at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
				at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
				at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
				at java.base/java.lang.Thread.run(Thread.java:833)
			Caused by: io.trino.spi.TrinoException: line 1:15: Table 'bigquery.tpch.tmp_trino_3a3f1a6f_e3d45e3f' does not exist
				at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
				at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:2179)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTable(StatementAnalyzer.java:484)
				at io.trino.sql.tree.Table.accept(Table.java:60)
				at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:501)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeFrom(StatementAnalyzer.java:4466)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:2933)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuerySpecification(StatementAnalyzer.java:484)
				at io.trino.sql.tree.QuerySpecification.accept(QuerySpecification.java:155)
				at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:501)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:509)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:1460)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitQuery(StatementAnalyzer.java:484)
				at io.trino.sql.tree.Query.accept(Query.java:107)
				at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
				at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:501)
				at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:463)
				at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:79)
				at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:71)
				at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:258)
				at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:196)
				at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:818)
				at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:149)
				at io.trino.$gen.Trino_testversion____20230320_122419_784.call(Unknown Source)
				at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
				at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
				at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
				at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
				at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
				at java.base/java.lang.Thread.run(Thread.java:833)
			]]
Expecting value to be true but was false
	at io.trino.testing.BaseFailureRecoveryTest.checkTemporaryTables(BaseFailureRecoveryTest.java:458)
	at io.trino.testing.BaseFailureRecoveryTest$FailureRecoveryAssert.cleansUpTemporaryTables(BaseFailureRecoveryTest.java:619)
	at io.trino.testing.BaseFailureRecoveryTest.testNonSelect(BaseFailureRecoveryTest.java:365)
	at io.trino.testing.BaseFailureRecoveryTest.testTableModification(BaseFailureRecoveryTest.java:342)
	at io.trino.testing.BaseFailureRecoveryTest.testTableModification(BaseFailureRecoveryTest.java:337)
	at io.trino.testing.BaseFailureRecoveryTest.testExplainAnalyze(BaseFailureRecoveryTest.java:315)
	at io.trino.testing.BaseFailureRecoveryTest$ParallelTestRunnable.run(BaseFailureRecoveryTest.java:899)
	at io.trino.testing.BaseFailureRecoveryTest.testParallel(BaseFailureRecoveryTest.java:222)
	at jdk.internal.reflect.GeneratedMethodAccessor1309.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:75)
	at org.testng.internal.TestMethodWithDataProviderMethodWorker.call(TestMethodWithDataProviderMethodWorker.java:14)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

colebow committed:
Add link to duration prop type in JDBC config fragment

sopel39 committed:
Add hive.metastore-stats-cache-ttl doc

findepi committed:
Remove redundant explicit session in test

findepi committed:
Use query runner without unnecessary tpch connector

findepi committed:
Fix query runner setup in TestWarnings

Set query runner's default session to point at the catalog/schema used
in the test, so that explicit session setup is not needed within test
body.

findepi committed:
Inline query runner setup in TestQueues

findepi committed:
Rename util class to indicate it's not a test

bitsondatadev commented:
@dprophet, could you take a look?

@yuuteng perhaps it makes sense to split the upgrade to 410 in a different branch? Let me know if we should meet to discuss the strategy here :)

findepi committed:
Use JDBC parameters in JDBC complex expression pushdown

findepi committed:
Disable testReadMetadataWithRelationsConcurrentModifications on H2 because of flakes

Under concurrently, H2 sometimes returns null table name in
`DatabaseMetaData.getTables`'s `ResultSet`.

hashhar committed:
Fix test class name in BigQuery cloud-tests profile

yuuteng committed:
Update to 411

yuuteng committed:
Update and fix errors

yuuteng committed:
Build successfully with 411

yuuteng committed:
Merge branch 'snowflake' of https://github.com/bloomberg/trino into bloomberg-snowflake

# Conflicts:
#	plugin/trino-snowflake/pom.xml
#	plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java
#	plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClientModule.java
#	plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeConfig.java
#	plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConfig.java
#	plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakePlugin.java

yuuteng committed:
Merge branch 'bloomberg-snowflake' into feature/2551-add-snowflake-connector

dprophet commented:
As discussed. This is a rebase with the master branch.
